src/components/Proposal/SentimentSurvey/SurveyResults.tsx

```diff
@@ -29,26 +29,22 @@ function initializeReactionCounters() {
   return reactionsCounters
 }
 
-function initializeTopicResults(surveyTopics: Pick<SurveyTopicAttributes, 'topic_id' | 'label'>[]) {
+function initializeTopicResults(surveyTopics: Topic[]) {
   const topicsResults: Record<any, Record<any, any>> = {}
   for (const topic of surveyTopics) {
-    topicsResults[topic.label] = initializeReactionCounters()
+    topicsResults[topic.topic_id] = initializeReactionCounters()
   }
   return topicsResults
 }
 
-function getResults(
-  surveyTopics: Pick<SurveyTopicAttributes, 'topic_id' | 'label'>[] | null,
-  votes: Record<string, Vote> | null
-) {
+function getResults(surveyTopics: Topic[] | null, votes: Record<string, Vote> | null) {
   if (!surveyTopics || !votes) return {}
-  const decoder = new SurveyDecoder(surveyTopics)
   const topicsResults = initializeTopicResults(surveyTopics)
   Object.keys(votes).map((key) => {
-    const survey: Survey = decoder.decode(votes[key].reason)
+    const survey: Survey = decodeSurvey(votes[key].metadata)
```

Comment:
If `decodeSurvey` returns a `Survey`, is typing this variable as a Survey necessary?